### ISSUE TITLE
Make http client configurable

### ISSUE THREAD

**Issue by vvraskin** [go-kivik/couchdb#153]
Opening it here since I'm using the couchdb driver, but maybe the change would require the driver interface update.

Currently the couchdb driver is using a standard http client [here](https://github.com/go-kivik/couchdb/blob/master/chttp/chttp.go#L60), but it would be good to make it configurable to allow some extra options for the transport.
Right now we are patching the client, but obviously this is not what we want to do long term. Would be good have it in the upstream. Is this something that could be also interesting for others?

**Comment by vvraskin** [comment]
cc @SugandhaFedUp

**Comment by flimzy** [comment]
That's a good suggestion.  I don't think it will make for a very intrusive change. Let me see what I can come up with.

**Comment by flimzy** [comment]
I would welcome your feedback on #154. This PR would allow specifying a configurable *http.Client, but by registering a new driver instance. Something like:

    kivik.Register("mycouch", &couchdb.Couch{
        HTTPClient: &http.Client{ ... },
    })

    client, err := kivik.New("mycouch", "http://localhost:5984/")

Would this suit your needs?  This would be very cumbersome if you needed an ever-changing HTTP client.  If that's your use case, I'd probably look at using the `Authenticate()` mechanism to provide a custom client, but that seems dirtier to me, unless a compelling case can be made.

What do you think?

**Comment by vvraskin** [comment]
Thank you for the quick feedback!
Yes, I like the approach, this should solve the issue - since we only need to specify the client once. 
Tomorrow I will have a closer look at the PR and get back to you.

**Comment by vvraskin** [comment]
@flimzy I'm currently testing the changes, do I need to do extra steps to support basic authentication after I specify the custom driver as in your example?

Previously I was providing the connection string like `http://user:host@localhost:5984/` and it worked, but now I receive unauthorized message. I will continue looking into that, just wanted to ask upfront, maybe you have a quick answer.

**Comment by flimzy** [comment]
Thanks for bringing that up.  I guess that might make sense. I'll take a look.

Are you setting the HTTP transport in your custom client?

**Comment by vvraskin** [comment]
I think I found the problem. It seems to be this peace of code that is ripping off the user and updates the dsnURL:
https://github.com/go-kivik/couchdb/blob/master/chttp/chttp.go#L58
However, I don't know why it worked before.
 
And I was able to fix it by specifying the basic authentication separately after I create the client:
```
	client.Authenticate(context.TODO(),couchdb.BasicAuth("user","pass"))
```
And this seems to be the right way to do that.
